test/e2e/framework/util.go

```diff
@@ -1052,6 +1052,25 @@ func WaitForPersistentVolumeClaimsPhase(phase v1.PersistentVolumeClaimPhase, c c
 	return fmt.Errorf("PersistentVolumeClaims %v not all in phase %s within %v", pvcNames, phase, timeout)
 }
 
+// findAvailableNamespaceName random namespace name starting with baseName.
+func findAvailableNamespaceName(baseName string, c clientset.Interface) (string, error) {
+	var name string
+	err := wait.PollImmediate(Poll, 30*time.Second, func() (bool, error) {
+		name = fmt.Sprintf("%v-%v", baseName, randomSuffix())
+		_, err := c.CoreV1().Namespaces().Get(name, metav1.GetOptions{})
+		if err == nil {
+			// Already taken
+			return false, nil
```

Comment:
I'm not following this. returning false, nil here, means we will retry again (and will probably get the same result). And in the end we will timeout.

I think you should return some kind of error here, which you will interpret differently outside.

Reply:
Returning false here means that we will retry and with high probability pick another name (name is generated in each loop separately).

Reply:
aah ok - we're generating name inside. Sorry. That makes sense now.